src/coord/src/session/vars.rs

```diff
@@ -113,6 +113,12 @@ const QGM_OPTIMIZATIONS: ServerVar<bool> = ServerVar {
     description: "Enables optimizations based on a Query Graph Model (QGM) query representation.",
 };
 
+const READ_THEN_WRITE_TIMEOUT: ServerVar<u64> = ServerVar {
+    name: static_uncased_str!("read_then_write_timeout"),
+    value: &10u64,
+    description: "Sets the timeout in seconds for INSERT...SELECT, UPDATE, and DELETE operations.",
+};
```

Comment:
Would love to avoid introducing a Materialize-specific session parameter here. I think you can use PostgreSQL's existing `idle_in_transaction_session_timeout` parameter here? Seems to be more or less exactly what we want.

Reply:
Also, these `*_timeout` session parameters are a bit funny. They take a string that can contain units:

```
benesch=# set idle_in_transaction_session_timeout = '1 minute';
ERROR:  invalid value for parameter "idle_in_transaction_session_timeout": "1 minute"
HINT:  Valid units for this parameter are "us", "ms", "s", "min", "h", and "d".
```

If you don't specify units, the implied unites are `ms`. We should definitely try to match PostgreSQL here, whatever name we choose for the parameter.

Reply:
Ah, Joe proposed a similar variable above and my apprehension with it is that to adhere to PG semantics, we need to plumb this timeout elsewhere (e.g. `implement_fast_path_peek`, which seems a little more complicated).

I'm down to modify the parsing to make the API consistent, but is there any appetite for getting this fix in before embarking on adding timeouts for all reads? Or are you comfortable with having the session variable's name have different semantics than it does in PG?

Reply:
I’m extremely in favor of using that parameter and just having slightly different semantics! Sorry, didn't mean to suggest doing more work to match the semantics exactly.